kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/DefaultKernelUtils.java

```diff
@@ -55,8 +58,25 @@ public static final MessageType pruneSchema(
      * @param field     Sub field given as Delta Kernel's {@link StructField}
      * @return {@link Type} of the Parquet field. Returns {@code null}, if not found.
      */
-    public static Type findSubFieldType(GroupType groupType, StructField field) {
-        // TODO: Need a way to search by id once we start supporting column mapping `id` mode.
+    public static Type findSubFieldType(
+            GroupType groupType,
+            StructField field,
+            Map<Integer, Type> parquetFieldIdToTypeMap) {
+
+        // First search by the field id. If not found, search by case-sensitive name. Finally
+        // by the case-insensitive name.
+        if (hasFieldId(field.getMetadata())) {
+            Integer deltaFieldId = getFieldId(field.getMetadata());
```

Comment:
I am actually inclined to everything in the field metadata as `string -> string` map type. Instead of `string -> object`. We followed the Delta-Spark model, but that may not be the right approach here.

cc @allisonport-db

Reply:
If we deserialize everything to a string won't that be an issue for writes?

Reply:
that is true. I think this is the artifact of now following the same schema for field metadata in the protocol.

Reply:
How does this all relate to https://github.com/delta-io/delta/blob/master/PROTOCOL.md#schema-serialization-format? I guess you're saying that spark's schema serialization uses string -> object map, so we can't really change that?

Reply:
Yep, we can't change that now.